### PR TITLE
Bug 1069611 - Set aria-hidden=true to top level elements when overlay is...

### PR DESCRIPTION
**Pull request by eeejay**
... visible.

### CODE REVIEW

**Comment by try-server-hook** [comment]
[eeejay Eitan Isaacson (eeejay)](https://github.com/eeejay) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=d4336f97513e)

File: apps/music/js/music.js [hunk 1]
```diff
@@ -128,7 +134,13 @@ var App = (function() {
...
     textElement.dataset.l10nId = l10nIds.text;
 
-    document.getElementById('overlay').classList.remove('hidden');
+    Array.forEach(document.body.children, function(elt) {
+      if (elt.id === 'overlay') {
+        elt.classList.remove('hidden');
+      } else {
+        elt.setAttribute('aria-hidden', true);
+      }
+    });
```
Comment (dominickuo) [inline apps/music/js/music.js]: @eeejay These two logics(Above and down) looks the same but opposite, do you think it's nice to use a function then pass different parameters? I am fine with the current code, just want to make these ui logics more organized.

File: tests/python/gaia-ui-tests/gaiatest/tests/accessibility/music/test_overlay.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    def test_empty_music(self):
+        # Requires there to be no songs on SDCard which is the default
+        music_app = Music(self.marionette)
+        music_app.launch()
+
+        music_app.wait_for_empty_message_to_load()
+
+        # Check that view elements are hidden when the overlay is showing.
+        self.assertTrue(self.accessibility.is_hidden(music_app.tabs))
+        self.assertTrue(self.accessibility.is_hidden(music_app.views))
```
Comment (dominickuo) [inline tests/python/gaia-ui-tests/gaiatest/tests/accessibility/music/test_overlay.py]: The tests looks good to me, but I think it's better to let QA to review this part as well, I will also set review request after I gave r+.

File: apps/music/js/music.js (hunk 1 above)
Comment (eeejay) [inline apps/music/js/music.js]: I'm not really sure if it is worth it. They are both in separate methods that are already there to show/hide the overlay, we might save 2-3 lines of code. I'll add comments above each of those loops.

**Comment by try-server-hook** [comment]
[eeejay Eitan Isaacson (eeejay)](https://github.com/eeejay) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=621705b4ff12)

**Comment by try-server-hook** [comment]
[eeejay Eitan Isaacson (eeejay)](https://github.com/eeejay) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=242fa789c36b)

File: tests/python/gaia-ui-tests/gaiatest/tests/accessibility/music/test_overlay.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from gaiatest import GaiaTestCase
+from gaiatest.apps.music.app import Music
+
+
+class TestOverlay(GaiaTestCase):
+
+    def setUp(self):
```
Comment (JohanLorenzo) [inline tests/python/gaia-ui-tests/gaiatest/tests/accessibility/music/test_overlay.py]: You don't need to define the `setUp` function here.

**Comment by try-server-hook** [comment]
[eeejay Eitan Isaacson (eeejay)](https://github.com/eeejay) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=4df296e8960e)

**Pull request merged**